lib/manageiq/providers/openstack/legacy/openstack_handle/image_delegate.rb

```diff
@@ -45,5 +45,13 @@ def images_for_accessible_tenants
       @os_handle.accessor_for_accessible_tenants(
         SERVICE_NAME, :images_with_pagination_loop, :id)
     end
+
+    def handle_upload(image, url)
+      file = open(url) if url =~ URI::DEFAULT_PARSER.make_regexp
```

Comment:
hm, this will do it in memory right? We should probably fetch it to /tmp and never hold the image data in memory.

Also this will be done by generic worker, so it will be killed in 10 minutes, so we'll need a separate longrunning job wrapped into a state machine.

cc @agrare can you point to some state machine examples? And if we have some code for handling big files. ^

Reply:
The ```open``` is expected to return TempFile for larger data files.

Reply:
Hey sorry I missed this.  :+1: for not loading the entire file into memory :smile:

We have a state machine for ansible-runner which will exec a process to do something and the state machine monitors it, https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/ansible_runner_workflow.rb

You could do something similar with wget -> /tmp and use a state machine similar to ^ to wait for it to exit.